subsys/usb/device/usb_device.c

```diff
@@ -776,8 +832,21 @@ static bool usb_set_interface(struct usb_setup_packet *setup)
 		case USB_DESC_ENDPOINT:
 			if (cur_iface == setup->wIndex) {
 				ep = (struct usb_ep_descriptor *)p;
-				ret = usb_eps_reconfigure(ep, cur_alt_setting,
-							  setup->wValue);
+
+				/*
+				 * only reconfigure endpoint if:
+				 * a) it hasn't previously been reconfigured; AND
```

Comment:
nitpick: a) is not indented the same as b) and c)

Reply:
Changed (the extra indent was actually intentional to show priority; but either way is OK...)

Reply:
Oh, I didn't notice that. Now, that you mentioned the priority, I see it. It essentially is `a) AND [b) or c)]`.
So maybe
```
    * only reconfigure endpoint if it hasn't previously been reconfigured AND
    * a) currently iterating over selected alt_setting; OR
    * b) endpoint not used in selected alt_setting
```
can be clearer. However, it is up to you.

Reply:
I like it - changed...